framework/src/executioners/Transient.C

```diff
@@ -729,3 +729,41 @@ Transient::setTimeStepper(TimeStepper & ts)
   mooseAssert(!_time_stepper, "Already set");
   _time_stepper = &ts;
 }
+
+bool
+Transient::convergedToSteadyState() const
+{
+  bool converged;
+
+  if (_check_aux)
+  {
+    // Get the relative change in the norm of each auxvariable
+    std::vector<Number> aux_soln_change_norms;
+    _aux.variableWiseRelativeSolutionDifferenceNorm(aux_soln_change_norms);
+
+    converged = true;
+    for (auto & norm_diff : aux_soln_change_norms)
+    {
+      // Normalize by timestep
+      norm_diff /= (_normalize_solution_diff_norm_by_dt ? _dt : Real(1));
+      if (norm_diff >= _steady_state_tolerance)
+      {
+        converged = false;
+        // No point in checking the rest of the auxvariables
```

Comment:
I m sure you can see how there can be so many different behaviors a user could want here and that's why it should be in a "Convergence" class rather than in this executioner (which is not the only executioner that does transient, see Griffin and its dozen or so executioners)

Reply:
this is just a comment. I m glad you're meeting the current user need here.